src/rgw/rgw_auth_s3.cc

```diff
@@ -700,13 +703,18 @@ std::string get_v4_string_to_sign(CephContext* const cct,
                                   const sha256_digest_t& canonreq_hash)
 {
   const auto hexed_cr_hash = buf_to_hex(canonreq_hash);
-  const auto string_to_sign = std::string()
+
+  /* We aren't adding the extra byte for the terminating 0 as it would be
+   * annihilated by the "-1" coming from hexed_cr_hash. */
```

Comment:
Hello @jmunhoz. Thanks for you review!

I've delegated the very-low-level things to dedicated functions (like the new variants of `buf_to_hex`, `calc_hmac_sha256`, `parse_key_value`, etc.) to avoid mixing them with the AWSv4 logic. Tracking such number of details on the business logic level seems overhelming to me.

The lifetime of objects typed as `boost::string_view` stays under assumption that `req_info` outlives the authentication phase.